src/commands/set.js

```diff
@@ -41,24 +46,32 @@ const writeConfigToFile = newConfig => {
 
 const checkBoolean = value => ['true', 'false'].includes(value);
 
-const setNestedConfigProperty = newValue => (obj, pathComponent, i, path) => {
-	if (i === path.length - 1) {
+const setNestedConfigProperty = newValue => (
+	obj,
+	pathComponent,
+	i,
+	dotNotationArray,
+) => {
+	if (i === dotNotationArray.length - 1) {
+		if (obj[pathComponent] === undefined) {
+			throw new ValidationError(writePropertyWarning(pathComponent));
```

Comment:
Maybe we need a new Error class: `ConfigurationError`?

Reply:
https://github.com/LiskHQ/lisky/issues/462